x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/IpFilterRemoteAddressFilterTests.java

```diff
@@ -60,6 +61,8 @@ public void init() throws Exception {
                     IPFilter.IP_FILTER_ENABLED_SETTING,
                     IPFilter.TRANSPORT_FILTER_ALLOW_SETTING,
                     IPFilter.TRANSPORT_FILTER_DENY_SETTING,
+                    TcpTransport.isUntrustedRemoteClusterEnabled() ? IPFilter.REMOTE_CLUSTER_FILTER_ALLOW_SETTING : null,
+                    TcpTransport.isUntrustedRemoteClusterEnabled() ? IPFilter.REMOTE_CLUSTER_FILTER_DENY_SETTING : null,
```

Comment:
Can you also extend the test to account for these new configuration ?

Reply:
The class this suite is testing didn't change and it's such a thin wrapper around `IPFilter` - it [delegates to it 100% of the time](https://github.com/elastic/elasticsearch/blob/12ad399c488f0cc60e19b5e1b29c6d569cb4351a/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/netty4/IpFilterRemoteAddressFilter.java#L31), so I'm not sure what benefit that would bring. Registering these settings is necessary if the feature flag is enabled though, because `IPFilter` tries to examine those settings on creation if the feature flag is enabled.

Reply:
I think it needs to be explicitly tested. By default, remote cluster port is not enabled which means the associated IP filtering won't be registered. Even after it is registered, we need update profileBoundAddresses to trigger rules being populated. I expanded the test class with a new test for IP filtering on the new port 36980e1584e959ab32a3999b55d158087361a201